src/main/java/com/rtb/projectmanagementtool/user/UserTaskChecklistServlet.java

```diff
@@ -0,0 +1,126 @@
+    auth = new AuthOps(datastore);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+
+    // Authenticate
+    Long userLoggedInId = auth.whichUserIsLoggedIn(request, response);
+    if (userLoggedInId == Long.parseLong(AuthOps.NO_LOGGED_IN_USER)) {
+      // If no user found, redirect to create user servlet
+      response.sendRedirect("/login");
+      return;
+    }
+
+    // new UserController
+    UserController userController = new UserController(datastore);
+
+    // get user by ID
+    if (request.getParameter("userID") == null) {
+      userID = auth.whichUserIsLoggedIn(request, response);
+    } else {
+      userID = Long.parseLong(request.getParameter("userID"));
+    }
+
+    UserData user = userController.getUserByID(userID);
+
+    // new TaskController
+    TaskController taskController = new TaskController(datastore);
+
+    // get this user's tasks
+    ArrayList<TaskData> tasks = taskController.getTasksByUserID(userID);
+
+    // build array to track color of task (red or green) and checked status
+    String[] color = new String[tasks.size()];
+    String[] checked = new String[tasks.size()];
+    for (int i = 0; i < tasks.size(); i++) {
+      TaskData task = tasks.get(i);
+      Status taskStatus = task.getStatus();
+      String statusString = taskStatus.name();
+      if (statusString.equals("COMPLETE")) {
+        color[i] = "green";
+        checked[i] = "checked";
+      } else {
+        color[i] = "red";
+        checked[i] = "";
+      }
+    }
+
+    // send task list, color array, and checked array
+    request.setAttribute("userTaskList", tasks);
+    request.setAttribute("color", color);
+    request.setAttribute("checkedStatus", checked);
+
+    // Load jsp for user page
+    request.getRequestDispatcher("task-checklist.jsp").forward(request, response);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+
+    // Authenticate
+    auth.loginUser(request, response);
+    Long userLoggedInId = auth.whichUserIsLoggedIn(request, response);
+
+    if (userLoggedInId == /*No user found*/ -1l) {
+      // If no user found, redirect to create user servlet
+      response.sendRedirect("/login");
+      return;
+    }
+
+    // new UserController
+    UserController userController = new UserController(datastore);
+
+    // get user by ID long
+    long userID = auth.whichUserIsLoggedIn(request, response);
+    UserData user = userController.getUserByID(userID);
+
+    // new TaskController
+    TaskController taskController = new TaskController(datastore);
+
+    // get this user's tasks
+    ArrayList<TaskData> tasks = taskController.getTasksByUserID(userID);
+
+    // get task status updates as strings
+    for (TaskData task : tasks) {
+      String taskStatusString = request.getParameter(task.getName());
```

Comment:
How is `task.getName()` getting the status? Also, same nit above can apply here.